### PR TITLE
Feature/decouple webapi from testexecution 2

### PR DESCRIPTION
**Pull request by jjelschen**
check #25 first!

### CODE REVIEW

File: src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/LocalSingleWorkerManager.xtend [hunk 1]
```diff
@@ -2,29 +2,49 @@ package org.testeditor.web.backend.testexecution.distributed.manager
...
 
 	override assign(WorkerInfo worker, TestJob job) {
-		if (worker === this.worker) {
-			this.worker.startJob(job).toCompletableFuture.get
+		return if (worker === this.worker) {
+			this.worker.startJob(job) => [
+				currentJob = job
+			]
+		} else {
+			//TODO throw exception
```
Comment (thrstnh) [inline src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/LocalSingleWorkerManager.xtend]: How about the comment? Throw an exception?

File: src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/TestExecutionManager.xtend [hunk 2]
```diff
@@ -48,4 +53,28 @@ class LocalSingleWorkerExecutionManager implements TestExecutionManager {
...
+	
+	override getStatusAll() {
+		return jobStore.statusAll + workerProvider.statusAll
+	}
+	
+	private def updateStatus(TestJobInfo job, TestStatus status) {
+		switch(status) {
+			case FAILED: job.setState(JobState.COMPLETED_WITH_ERROR)
+			case SUCCESS: job.setState(JobState.COMPLETED_SUCCESSFULLY)
+			default: job // TODO error / exception handling?
```
Comment (thrstnh) [inline src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/TestExecutionManager.xtend]: How about the comment about exception handling?

File: src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/LocalSingleWorkerManager.xtend (hunk 1 above)
Comment (jjelschen) [inline src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/LocalSingleWorkerManager.xtend]: … what it says in the TODO :smile:

File: src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/TestExecutionManager.xtend (hunk 2 above)
Comment (jjelschen) [inline src/main/java/org/testeditor/web/backend/testexecution/distributed/manager/TestExecutionManager.xtend]: implement TODO

**Review by thrstnh — APPROVED** [review]

**Pull request merged**